homeassistant/components/sonos/entity.py

```diff
@@ -57,10 +57,11 @@ async def async_will_remove_from_hass(self) -> None:
     async def async_fallback_poll(self, now: datetime.datetime) -> None:
         """Poll the entity if subscriptions fail."""
         if not self.speaker.subscriptions_failed:
+            listener_msg = self.speaker.subscription_address
             if soco_config.EVENT_ADVERTISE_IP:
-                listener_msg = f"{self.speaker.subscription_address} (advertising as {soco_config.EVENT_ADVERTISE_IP})"
-            else:
-                listener_msg = self.speaker.subscription_address
+                listener_msg = (
+                    f"{listener_msg} (advertising as {soco_config.EVENT_ADVERTISE_IP})"
+                )
             message = f"{self.speaker.zone_name} cannot reach {listener_msg}, falling back to polling, functionality may be limited"
```

Comment:
If `speaker.subscription_address` is changed to only ever return a `str`, I don't think that needs to change.